libraries/psio/include/psio/reflect.hpp

```diff
@@ -0,0 +1,760 @@
+      static constexpr bool isConstFunction = false;
+      using ValueType                       = V;
+   };
+
+   template <typename R, typename T, typename... Args>
+   struct MemberPtrType<R (T::*)(Args...)>
+   {
+      static constexpr bool isFunction      = true;
+      static constexpr bool isConstFunction = false;
+      static constexpr int  numArgs         = sizeof...(Args);
+      using ClassType                       = T;
+      using ReturnType                      = R;
+      using ArgTypes                        = TypeList<Args...>;
+      using SimplifiedArgTypes              = TypeList<std::remove_cvref_t<Args>...>;
+   };
+
+   template <typename R, typename T, typename... Args>
+   struct MemberPtrType<R (T::*)(Args...) const>
+   {
+      static constexpr bool isFunction      = true;
+      static constexpr bool isConstFunction = true;
+      static constexpr int  numArgs         = sizeof...(Args);
+      using ClassType                       = T;
+      using ReturnType                      = R;
+      using ArgTypes                        = TypeList<Args...>;
+      using SimplifiedArgTypes              = TypeList<std::remove_cvref_t<Args>...>;
+   };
+
+   // Not really a member, but usable via std::invoke
+   template <typename R, typename T, typename... Args>
+   struct MemberPtrType<R (*)(T&, Args...)>
+   {
+      static constexpr bool isFunction      = true;
+      static constexpr bool isConstFunction = false;
+      static constexpr int  numArgs         = sizeof...(Args);
+      using ClassType                       = T;
+      using ReturnType                      = R;
+      using ArgTypes                        = TypeList<Args...>;
+      using SimplifiedArgTypes              = TypeList<std::remove_cvref_t<Args>...>;
+   };
+
+   // Not really a member, but usable via std::invoke
+   template <typename R, typename T, typename... Args>
+   struct MemberPtrType<R (*)(const T&, Args...)>
+   {
+      static constexpr bool isFunction      = true;
+      static constexpr bool isConstFunction = true;
+      static constexpr int  numArgs         = sizeof...(Args);
+      using ClassType                       = T;
+      using ReturnType                      = R;
+      using ArgTypes                        = TypeList<Args...>;
+      using SimplifiedArgTypes              = TypeList<std::remove_cvref_t<Args>...>;
+   };
+
+   template <typename F, typename... Args>
+   void forEachType(TypeList<Args...>, F&& f)
+   {
+      (f((remove_cvref_t<Args>*)nullptr), ...);
+   }
+
+   template <typename F, typename... Args>
+   void forEachNamedType(TypeList<Args...>, std::span<const char* const> names, F&& f)
+   {
+      size_t i = 0;
+      auto   g = [&](auto* p)
+      {
+         f(p, i < names.size() ? names.begin()[i] : nullptr);
+         ++i;
+      };
+      (g((remove_cvref_t<Args>*)nullptr), ...);
+   }
+
+}  // namespace psio
+
+// TODO: not legal to add new definitions to std namespace
```

Comment:
Why is it needed in the first place? There's an overload for `string` in `get_type_name.hpp`